src/Installers/Windows/TargetingPack/Product.wxs

```diff
@@ -5,7 +5,13 @@
         <Package InstallerVersion="$(var.InstallerVersion)" Compressed="yes" InstallScope="perMachine" />
 
         <MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed." Schedule="afterInstallFinalize" />
-        <Media Id="1" Cabinet="$(var.Cabinet)" CompressionLevel="high" EmbedCab="$(var.EmbedCab)" />
+
+        <?if $(var.EmbedCab)=yes?>
+            <!-- Ignore var.Cabinet This element should choose an appropriate name for the embedded CAB. -->
+            <MediaTemplate CompressionLevel="high" EmbedCab="yes" />
```

Comment:
Is an `id` attribute needed in this element❔

Reply:
No, the MediaTemplate is just a convenience element in the toolset that defers some of the mundane tasks to the compiler/linker when generating the table.

Reply:
Do you think ASP.NET would move back to external cabinets? If not, you can likely just use the MediaTemplate. The downside is that with external cabinets and the template and multiple MSIs is that they can't exist in the same folder (becuase the cabs are just numerically named and will clash).

Reply:
I doubt it but prefer to leave the option available without needing additional work to avoid those clashes.